app/helpers/custom_components_helper.rb

```diff
@@ -0,0 +1,22 @@
+module CustomComponentsHelper
+  def raw_with_custom_components(stringish)
+    fragment = Nokogiri::HTML.fragment(stringish)
+
+    fragment.xpath('.//embedded-app').each do |f|
```

Comment:
Ja by som mozno pouzil radsej `fragment.css('embedded-app')`.
`fragment.xpath('.//embedded-app')` najde len root elementy, a ak niekto to pastne vnorene v nejakom elemente, uz tvoj `xpath` to nenajde.

Reply:
`fragment.xpath('.//embedded-app')` podla mna najde aj vnorene, nie?

Reply:
Som tiez za CSS, xpath uz dnes nikto nepozna :)

Reply:
Este taky detail.
Co sa stane ak niekto pouzije invalid html? 

Neuzatvoreny tag napr. Necrashne to?

Reply:
Crashnut sa mi to nepodarilo. Akurat sa to mozno dakedy zachova inak ako by sa s tym vysporiadal browser po obycajnom `raw` (ale pri invalid html je to podla mna toto riziko OK).